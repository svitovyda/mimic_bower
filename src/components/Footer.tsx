import * as React from "react";

export const Footer: React.FC = () => {
  return <div></div>;
};
