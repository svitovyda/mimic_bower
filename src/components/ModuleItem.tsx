import type { Package } from "../models/Package";
import * as React from "react";

export interface ModuleItemProp {
  packageElement: Package;
}

export const ModuleItem: React.FC<ModuleItemProp> = ({ packageElement }: ModuleItemProp) => {
  return (
    <div>
      {packageElement.name} {packageElement.owner} {packageElement.stars}
    </div>
  );
};
