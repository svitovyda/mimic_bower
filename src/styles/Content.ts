import styled from "styled-components";

export const ContentContainer = styled.div`
  border-radius: 2px;
  box-shadow: 0 0 1px #ccc;
  padding: 10px;
`;

export const ContentErrorContainer = styled.div`
  padding: 40px 0;
  text-align: center;
`;
