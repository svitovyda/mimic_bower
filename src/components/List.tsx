import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
