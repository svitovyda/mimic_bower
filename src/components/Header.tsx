import { HeaderContainer, HeaderLogo, HeaderSubtitle, HeaderTitle } from '../styles/Header';
import * as React from 'react';

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{' '}
          <a href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </a>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
