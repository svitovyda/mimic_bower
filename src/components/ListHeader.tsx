/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
