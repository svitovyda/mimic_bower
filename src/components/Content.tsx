import { useContentFetcher } from "../hooks/contentFetcher";
import { usePagination } from "../hooks/usePagination";
import type { Package } from "../models/Package";
import { List } from "./List";
import { Pagination } from "./Pagination";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import configJson from "config";
import * as React from "react";

const ItemsPerPage = configJson.searchItemsPageMaxSize;

export interface ContentProps {
  query: string;
}

export const Content: React.FC<ContentProps> = ({ query }: ContentProps) => {
  const [fetchState, setQuery] = useContentFetcher("");
  const [paginationState, navigateToPage] = usePagination(fetchState.data);
  const [page, setPage] = React.useState<Package[]>([]);

  React.useEffect(() => {
    setQuery(query);
  }, [query, setQuery]);

  React.useEffect(() => {
    setPage(
      fetchState.data.slice(
        (paginationState.currentPage - 1) * ItemsPerPage,
        paginationState.currentPage * ItemsPerPage
      )
    );
  }, [fetchState.data, paginationState]);

  if (fetchState.isLoading) return <LoaderAnimation />;
  if (fetchState.isError) return <div>Some error occurred, try again later!</div>;
  return (
    <div>
      <List data={page} />
      <Pagination
        currentPage={paginationState.currentPage}
        totalPages={paginationState.totalPages}
        navigateToPage={navigateToPage}
      />
    </div>
  );
};
