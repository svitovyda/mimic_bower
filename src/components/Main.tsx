import { Content } from "./Content";
import { Search } from "./Search";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <div>
      <Search onQueryChanged={onNewQuery} />
      <Content query={query} />
    </div>
  );
};
