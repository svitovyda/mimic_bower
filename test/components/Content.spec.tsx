import { Content } from "../../src/components/Content";
import { render } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";

describe("Content", () => {
  it("renders without error", () => {
    const component = render(<Content query="" />);

    expect(component).toMatchSnapshot();
  });
});
