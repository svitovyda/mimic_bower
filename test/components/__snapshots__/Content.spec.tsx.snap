// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  text-align: center;
  background-color: white;
  border: 2px solid #007aff;
  padding: 6px 12px;
  font-size: 16px;
  border-radius: 0.25rem;
  color: #008ec4;
}

.c0:hover:enabled {
  color: #fff;
  background-color: #0d6efd;
  border-color: #0d6efd;
}

.c0:disabled {
  color: lightgray;
  border-color: lightgray;
}

<body>
    <div>
      <div>
        <div />
        <div>
          <button
            class="c0"
            disabled=""
          >
            ← Previous
          </button>
          <button
            class="c0"
            disabled=""
          >
            Next →
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  text-align: center;
  background-color: white;
  border: 2px solid #007aff;
  padding: 6px 12px;
  font-size: 16px;
  border-radius: 0.25rem;
  color: #008ec4;
}

.c0:hover:enabled {
  color: #fff;
  background-color: #0d6efd;
  border-color: #0d6efd;
}

.c0:disabled {
  color: lightgray;
  border-color: lightgray;
}

<div>
    <div>
      <div />
      <div>
        <button
          class="c0"
          disabled=""
        >
          ← Previous
        </button>
        <button
          class="c0"
          disabled=""
        >
          Next →
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
